proposed/2022/dotnet-nuget-why-proposal.md

```diff
@@ -0,0 +1,137 @@
+- `project.assets.json` file that maintains a project's dependency graph, which is used to make sure that all necessary packages are installed on the computer.
+- `{projectName}.nuget.dgspec.json` file that maintains a project's top-level dependencies along with other metadata.
+- `{projectName}nuget.g.targets` and `{projectName}.nuget.g.props` files define various properties, such as the path to where the packages cache is on the local machine. They also include any MSBuild imports that NuGet packages referenced by the project need.
+
+The contents of `{projectName}.nuget.dgspec.json` and `project.assets.json` will be helpful to understand the dependency graph of a given package but there are few limitations.
+
+-  NuGet restore operation downloads packages during dependency resolution which may not part of the final graph The absence of a package id and version combination in `project.assets.json` file signals that it was downloaded during dependency resolution but not part of the final dependency graph.
+- Packages acquired through `PackageDownload` are not tied to the project in any way beyond acquisition. These packages are not recorded in the final graph that `project.assets.json` maintains.
+
+`dotnet nuget why` command prints the dependency graph of a given package only if it is part of the final graph.
+
+```
+dotnet nuget why [<PROJECT>|<SOLUTION>] <PACKAGE_NAME>
+    [-f|--framework <FRAMEWORK>]
+    [-v|--verbosity <LEVEL>]
+
+dotnet nuget why -h|--help
+```
+#### Arguments
+
+- PROJECT | SOLUTION
+
+The project or solution file to operate on. If not specified, the command searches the current directory for one. If more than one solution or project is found, an error is thrown.
+
+- PACKAGE_NAME
+
+The package for whom the dependency graph has to be identified. Package name wildcards are not supported.
+
+#### Options
+
+- -f|--framework <FRAMEWORK>
+
+Search package depdnency graph only for a specific target framework.
+
+- -?|-h|--help
+
+Prints out a description of how to use the command.
+
+- -v|--verbosity <LEVEL>
+
+Sets the verbosity level of the command. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic]. The default is minimal. For more information, see [LoggerVerbosity](https://docs.microsoft.com/en-us/dotnet/api/microsoft.build.framework.loggerverbosity).
+
+#### Examples
+
+- List dependency graph of a package given package id.
+
+```
+dotnet nuget why packageA
+
+Project 'projectNameA' has the following dependency graph for 'packageA'
+   [net6.0]: Microsoft.ML (1.0.0) > Microsoft.ML.Util (1.0.0) > packageA (1.0.0)
+   [net472] Microsoft.ML (1.0.0) > Microsoft.ML.Util (1.0.0) > packageA (1.0.0)
+
+Project 'projectNameB' has the following dependency graph for 'packageA'
+   [net6.0]: Microsoft.ML (1.1.0) > Microsoft.ML.Util (1.1.0) > packageA (1.1.0)
+   [net472] Microsoft.ML (1.1.0) > Microsoft.ML.Util (1.1.0) > packageA (1.1.0)
```

Comment:
Personally, I feel `->` as separator is more readable for end users. What do you think?
![image](https://user-images.githubusercontent.com/8766776/176038051-6641e243-d00c-4c77-a311-bbeb9dd4bae3.png)

Reply:
That's a good point, I agree!